### DISCUSSION TITLE
quarto render website

### DISCUSSION THREAD

**Discussion by kbvernon** [posit-dev/positron#4428] [Q&A] [answered]
I am struggling to preview/render a Quarto website in Positron with the Quarto extension installed. Near as I can tell, there is only an option to preview/render at the level of individual documents (with `Ctrl + Shift + K`), not entire projects. I also tried `quarto preview <name of project dir>` in the terminal, but got the "could not find" error. Maybe user error?

Anyway, I'm enjoying Positron a lot. Just looking for guidance here. Thanks!

**Comment by juliasilge** [comment, marked as answer]
Previewing a website _does_ work for me, both from the button like this (this will preview the whole site, not just the single document):

![preview-site](https://github.com/user-attachments/assets/7e6d4389-a867-4c81-a7b6-0cf041559ee8)

Or from the terminal in Positron like `quarto preview`. Can you say more about the errors you are seeing?

You can read more here: https://quarto.org/docs/websites/#website-preview

**Reply by kbvernon** [reply]
Another #RTFM moment! I'm really on a roll today. Sorry again!

Yes, I see now that `quarto preview` does render the whole site. I was under the impression that it was working like Render in RStudio and only showing a specific page that I was actively editing. In fact, it renders the whole site, but opens the preview on the page that is currently being edited. Slightly different behavior then build, which opens a preview on the home page by default. 

I guess I should mention user error, too: I didn't realize that `quarto preview mysite` assumes that mysite is a sub-directory of whatever directory you're currently in. 

Thanks for the quick reply here!

Also, for future reference, should anything Quarto related be discussed at the Quarto repo? I just assumed - maybe naively - that because I was working in Positron, it was worth posting here.

**Reply by juliasilge** [reply]
It is not always clear to folks coming to Quarto and Positron what behavior is coming from where, so no worries on that! FWIW I would expect that _most_ things you observe are more about Quarto behavior than Positron behavior.
